### PR TITLE
Add more test for OpenAPI Specification v3.1

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**

### CODE REVIEW

File: springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app8.json [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "ExamplesResponse": {
```
Comment (JohnNiang) [inline springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app8.json:37]:
```suggestion
      "ExamplesResponse": {
        "type": "object",
```

**Review by JohnNiang — COMMENTED** [review]
IMO, schema type is missing here.

File: springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app8.json (hunk 1 above)
Comment (xiaoxiangmoe) [reply springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app8.json:37]: @JohnNiang It seems that the current unit test also not include `"type": "object"`

https://github.com/springdoc/springdoc-openapi/blob/d4f99ac9036a00ade542c3ebb1ce7780a171d7d8/springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app7.json#L37-L60

Comment (xiaoxiangmoe) [reply springdoc-openapi-starter-webmvc-api/src/test/resources/results/3.1.0/app8.json:37]: See here: https://spec.openapis.org/oas/v3.1.0#data-types

> Data types in the OAS are based on the types supported by the [JSON Schema Specification Draft 2020-12](https://tools.ietf.org/html/draft-bhutton-json-schema-00#section-4.2.1).

In JSON Schema Specification Draft 2020-12, "type" field is optional, so this is valid JSON schema.

And the api-gen is done by swagger-core. We can advise swagger-core to provide `"type": "object"`.

**Comment by xiaoxiangmoe** [comment]
@uc4w6c Can you review it?

**Comment by uc4w6c** [comment]
@xiaoxiangmoe 
Sorry for the late confirmation.
Since I don't have merge privileges, I requested bnasslahsen to be a reviewer.

I thought it might be better to combine it with the test code below.
https://github.com/springdoc/springdoc-openapi/tree/main/springdoc-openapi-starter-webmvc-api/src/test/java/test/org/springdoc/api/v31/app7

**Comment by bnasslahsen** [comment]
@xiaoxiangmoe, 

Thank you for your contribution to the project.

**Pull request merged**